youtube_dl/extractor/pornhub.py

```diff
@@ -46,6 +50,62 @@ def dl(*args, **kwargs):
 
         return webpage, urlh
 
+    def _login_if_required(self, host):
+        login_info = self._get_login_info(netrc_machine=host.split('.')[0])
+
+        # PornHub Premium requires some kind of authentication
+        if 'premium' in host:
+            cookie_file = self._downloader.params.get('cookiefile')
+            if not cookie_file and not all(login_info):
+                raise ExtractorError(
+                    'PornHub Premium requires authentication.'
+                    ' You may want to use --cookies or --netrc.',
+                    expected=True)
+
+        cookies = self._get_cookies('https://%s' % host)
+        if all(login_info) and not cookies:
+            self._login(host, login_info)
```

Comment:
1. Having arbitrary cookies set for domain does not mean user is logged in so this will potentially skip login when it should not.
2. `cookies` will always be non empty since `age_verified` cookie is always set by the extractor.

Reply:
This was an attempt to avoid submitting multiple "log-in" status checks when downloading playlists. I will remove as I can see it's causing inconsistent behavior.